tasks/config.yml

```diff
@@ -238,6 +238,13 @@
   - restart freeswitch
   - reload systemd
 
+- name: configure freeswitch to use correct IP address for wss-binding
+  lineinfile:
+    path: /opt/freeswitch/conf/sip_profiles/external.xml
+    regexp: "<param name=\"wss-binding\"  value=\":7443\"/>"
+    line: "    <param name=\"wss-binding\"  value=\"{{ ansible_default_ipv4.address }}:7443\"/>"
```

Comment:
Setting this default to `{{ ansible_default_ipv4.address }}` might lead to future problems.
Example would be some mastermind who is only using IPv6. 😬
Another example would be users with multible public IPs.

IMO:
Replacing `{{ ansible_default_ipv4.address }}` with a default variable `{{ freeswitch_ip_address }}` in `defaults/main.yml` `freeswitch_ip_address: "{{ ansible_default_ipv4.address }}"` would be better.

This way it still defaults to `{{ ansible_default_ipv4.address }}` but there would be an option to overwrite it.
When doing this also the README needs an update and explanation when this is used.

Reply:
I didn't see using `{{ ansible_default_ipv4.address }}` as a problem first since it was already used in the nginx configuration - but you're right.

`{{ freeswitch_ip_address }}` sounds very general so I looked into where to set the IP address for freeswitch properly and found that: [](https://freeswitch.org/confluence/display/FREESWITCH/local_ip_v4)
Sadly, for some reason this breaks the ICE negotation for us in certain network situations.

So I would stay with only changing the IP address for the wss-binding but replace it with a default variable as suggested and update the README accordingly.

Reply:
Good point!
Then NGINX config must be optimized as well.